src/scheduler/node_info.cpp

```diff
@@ -2383,11 +2383,8 @@ eval_selspec(status *policy, selspec *spec, place *placespec,
 	/* Remove visited, scattered and ineligible bits from ncsr for node searching
 	 * Since statebusy is a per cycle bit, it shouldn't be changed
 	 */
-	for (i = 0; ninfo_arr[i] != NULL; i++) {
-		ninfo_arr[i]->nscr &= ~NSCR_VISITED;
-		ninfo_arr[i]->nscr &= ~NSCR_SCATTERED;
-		ninfo_arr[i]->nscr &= ~NSCR_INELIGIBLE;
-	}
+	for (i = 0; ninfo_arr[i] != NULL; i++)
+		ninfo_arr[i]->nscr &= ~(NSCR_VISITED | NSCR_SCATTERED | NSCR_INELIGIBLE);
```

Comment:
Can we not make a macro value of all these three flags together? That would mean the scheduler will not be doing this or operation every time for every job.

Reply:
i doubt bitwise operations cost much, these are in an enum, so making a macro for them might be a little weird .. @bhroam  what do you think about @arungrover 's idea?

Reply:
I tried checking the assembly with or without macro and there is no difference at all. So I take it back, It's fine the way it is now.